### ISSUE TITLE
GameBase.MinimizedMinimumUpateRate & ThreadThrottler appare to have issues/ Can't enforce minimum update rate on minimized games.

### ISSUE THREAD

**Issue by devjarmo** [stride3d/stride#403]
**Release Type**: Official Release

**Version**: 3.0

**Platform(s)**: Windows

**Describe the bug**
When setting MinimizedMinimumUpdateRate and/or changing the throttle type to any option Standard, PreciseAuto or PreciseManual the game appears to sleep indefinitely when minimized. The throttle does not appear to be effective in enforcing a minimum rate. This is an issue for locally hosted multiplayer games when the server is minimized, as well as headless implementations when the server was minimized.

**To Reproduce**
1. Create a game
2. Try to modify ((GameBase)Game).MinimizedMinimumUpdateRate
3. Optionally change the type using the set methods.
4. Minimize the window

Note: You will likely need to put something into the game so that you are aware of game rate when minimized.

**Expected behavior**
I would expect setting the MinimizedMinimumUpdateRate would throttle or rate limit the game to enforce a minimum rate, in this case I don't want any limit. I'd request an option be added to the Throttler for Disabled and a SetDisabled() method be added. Optionally this could be added to GameBase to prevent the throttle from ever being checked in the if statement for if Minimized check.

**Screenshots**
If applicable, add screenshots to help explain your problem (use GitHub drag & drop)

**Log and callstacks**
If any, please attach here any log or callstack (preferably in a `.txt` file using GitHub drag and drop)

**Additional context**
I looked into the source and I'm not sure I'm seeing what I'm actually running locally; one type appears to not call the method in the throttler (SetToPreciseAuto). Also, I don't have the time at the moment to debug the sleep logic.

**Comment by tebjan** [comment]
should be a 5 minute action for @Eideren :)

**Comment by Eideren** [comment]
I cannot reproduce this issue: spawned a new animation sample, changed UIScript to 
```
using System;
using System.Threading.Tasks;
using Xenko.Engine;
using Xenko.Games;
using Xenko.Graphics;
using Xenko.UI;
using Xenko.UI.Controls;

namespace AnimatedModel3
{
    public class UIScript : AsyncScript
    {
        public Entity Knight;

        public SpriteFont Font;

        public override async Task Execute()
        {
            // Bind the buttons
            var page = Entity.Get<UIComponent>().Page;

            var btnIdle = page.RootElement.FindVisualChildOfType<Button>("ButtonIdle");
            btnIdle.Click += (s, e) => Knight.Get<AnimationComponent>().Crossfade("Idle", TimeSpan.FromSeconds(0.25));

            var btnRun = page.RootElement.FindVisualChildOfType<Button>("ButtonRun");
            btnRun.Click += (s, e) => Knight.Get<AnimationComponent>().Crossfade("Run", TimeSpan.FromSeconds(0.25));

            // Set the default animation
            Knight.Get<AnimationComponent>().Play("Run");
            int frames = 0;
            // One frame every 30 ms
            ((GameBase)Game).MinimizedMinimumUpdateRate.MinimumElapsedTime = TimeSpan.FromMilliseconds( 30d );
            while (Game.IsRunning)
            {
                await Script.NextFrame();
                frames++;
                page.RootElement.FindVisualChildOfType<TextBlock>().Text = $"{frames} - {Game.UpdateTime.Elapsed.Milliseconds}";  
            }
        }
    }
}
```
``frames`` wasn't stalled while the game was minimized, please create a repro project !

To avoid any blocking you just have to set ``MinimumElapsedTime`` to zero: you allow the time between each frame to be zero or greater, you'll never have negative time between frames so no blocking will occur. Granted I haven't made the API as obvious as I could have in retrospect.

**Comment by DevJarmo** [comment]
Thank you, I will attempt to set MinimimumElapsedTime to zero and test. If issue persists I will create a repo.

Would it change anything if you were using a SyncScript? I am currently using SyncScripts.

**Comment by Eideren** [comment]
Nope, changed the class to inherit from SyncScript, converted the code to the SyncScript pattern and the result's pretty much the same.

**Comment by DevJarmo** [comment]
Thanks for testing that, I will take a look at my source tonight. I also had TreatFocusedLikeMinimized set to false and a couple of other parameters set, maybe one of those is contributing.

**Comment by DevJarmo** [comment]
Again thanks for looking at this issue, this is what I've found:

1) When I do not set MinimumElapsedTime the game loop halts when minimized.
2) If I set it to TimeSpan.Zero it runs as fast as possible ~480 ticks per second.
3) When I set it to the same rate as the game loop (1/60f) it is stable and works while minimized.
    a) This is the same behavior with TreatNotFocusedLikeMinimized is true or false.

Makes sense in hindsight, this is what I would expect. I believe before I was confused with the different Set Precise methods.

I will leave this open to be closed by another; in case anyone wants to add any clarification to the ticket or update the source with a clearer language--Perhaps something like SustainGameRateWhileMinimized or something.
